### PR TITLE
force removal of python api to guarantee version install

### PR DESCRIPTION
**Pull request by n-hutton**
I usually run the install dependencies before running the E2E tests to ensure a version match. However this appears to occasionally silently fail when there is already a version installed

### CODE REVIEW

**Review by cyenyxe — DISMISSED** [review]
You may have to grep the output of `pip list` or `pip show` before trying to uninstall, because pip fails when the package is not installed.

Could you also state the reason for the uninstall in the bash script itself? Otherwise in the future when we don't remember, it may seem unnecessary and someone will delete it 😨

File: scripts/install-test-dependencies.sh [hunk 1]
```diff
@@ -1,6 +1,8 @@
 #!/bin/bash
 set -e
 
+pip3 uninstall -y fetchai-ledger-api
```
Comment (ejfitzgerald) [inline scripts/install-test-dependencies.sh]: Do we understand why `--force-resintall` is not covering this case. Honestly, I am not sure that I have run into this issue but maybe I am special! 😁

**Comment by n-hutton** [comment]
@cyenyxe @ejfitzgerald Perhaps I misunderstand something, but my behaviour is as follows:

```
$ pip3 freeze | grep fetch
fetchai-ledger-api==0.10.2
fetchai-netutils==0.0.6a1

$ cat ./scripts/install-test-dependencies.sh
#!/bin/bash
set -e

pip3 install --no-cache-dir --ignore-installed --force-reinstall wheel pyyaml requests
pip3 install --no-cache-dir --ignore-installed --force-reinstall fetchai-ledger-api==0.10.1
pip3 install --no-cache-dir --ignore-installed --force-reinstall ./scripts/fetchai_netutils

$ ./scripts/install-test-dependencies.sh
Collecting wheel
  Downloading https://files.pythonhosted.org/packages/00/83/b4a77d044e78ad1a45610eb88f745be2fd2c6d658f9798a15e384b7d57c9/wheel-0.33.6-py2.py3-none-any.whl
Collecting pyyaml
  Downloading https://files.pythonhosted.org/packages/8d/c9/e5be955a117a1ac548cdd31e37e8fd7b02ce987f9655f5c7563c656d5dcb/PyYAML-5.2.tar.gz (265kB)
     |████████████████████████████████| 266kB 6.3MB/s
Collecting requests
  Downloading https://files.pythonhosted.org/packages/51/bd/23c926cd341ea6b7dd0b2a00aba99ae0f828be89d72b2190f27c11d4b7fb/requests-2.22.0-py2.py3-none-any.whl (57kB)
     |████████████████████████████████| 61kB 9.1MB/s
Collecting chardet<3.1.0,>=3.0.2
  Downloading https://files.pythonhosted.org/packages/bc/a9/01ffebfb562e4274b6487b4bb1ddec7ca55ec7510b22e4c51f14098443b8/chardet-3.0.4-py2.py3-none-any.whl (133kB)
     |████████████████████████████████| 143kB 4.5MB/s
Collecting idna<2.9,>=2.5
  Downloading https://files.pythonhosted.org/packages/14/2c/cd551d81dbe15200be1cf41cd03869a46fe7226e7450af7a6545bfc474c9/idna-2.8-py2.py3-none-any.whl (58kB)
     |████████████████████████████████| 61kB 20.5MB/s
Collecting urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1
  Downloading https://files.pythonhosted.org/packages/b4/40/a9837291310ee1ccc242ceb6ebfd9eb21539649f193a7c8c86ba15b98539/urllib3-1.25.7-py2.py3-none-any.whl (125kB)
     |████████████████████████████████| 133kB 7.6MB/s
Collecting certifi>=2017.4.17
  Downloading https://files.pythonhosted.org/packages/b9/63/df50cac98ea0d5b006c55a399c3bf1db9da7b5a24de7890bc9cfd5dd9e99/certifi-2019.11.28-py2.py3-none-any.whl (156kB)
     |████████████████████████████████| 163kB 5.8MB/s
Building wheels for collected packages: pyyaml
  Building wheel for pyyaml (setup.py) ... done
  Created wheel for pyyaml: filename=PyYAML-5.2-cp37-cp37m-macosx_10_14_x86_64.whl size=151495 sha256=ef54710d529c2bc997e7f18b12602da326dc5d2f1c7c330ca46791671715bdde
  Stored in directory: /private/var/folders/7r/f8pztdh56yl6hd3mjzyc8cp40000gn/T/pip-ephem-wheel-cache-aa7majek/wheels/54/b7/c7/2ada654ee54483c9329871665aaf4a6056c3ce36f29cf66e67
Successfully built pyyaml
Installing collected packages: wheel, pyyaml, chardet, idna, urllib3, certifi, requests
Successfully installed certifi-2019.11.28 chardet-3.0.4 idna-2.8 pyyaml-5.2 requests-2.22.0 urllib3-1.25.7 wheel-0.33.6
Collecting fetchai-ledger-api==0.10.1
  Downloading https://files.pythonhosted.org/packages/95/f4/caf84be89e88764e5969b2661512d12c4a9c43dc8259e54f5e6b689b4540/fetchai_ledger_api-0.10.1-py3-none-any.whl (78kB)
     |████████████████████████████████| 81kB 5.1MB/s
Collecting requests
  Downloading https://files.pythonhosted.org/packages/51/bd/23c926cd341ea6b7dd0b2a00aba99ae0f828be89d72b2190f27c11d4b7fb/requests-2.22.0-py2.py3-none-any.whl (57kB)
     |████████████████████████████████| 61kB 10.2MB/s
Collecting ecdsa
  Downloading https://files.pythonhosted.org/packages/a2/25/3bb32da623b39a27a07d194cd58e4540224421d924661de2e694304ae4fa/ecdsa-0.14.1-py2.py3-none-any.whl (79kB)
     |████████████████████████████████| 81kB 3.9MB/s
Collecting base58
  Downloading https://files.pythonhosted.org/packages/09/b2/21ac9591f055acc145afead895edeb73bbd69d95cf366fc5c2233f2434cb/base58-1.0.3-py3-none-any.whl
Collecting pyaes
  Downloading https://files.pythonhosted.org/packages/44/66/2c17bae31c906613795711fc78045c285048168919ace2220daa372c7d72/pyaes-1.6.1.tar.gz
Collecting msgpack
  Downloading https://files.pythonhosted.org/packages/68/a6/18d0f681fa650a3ff10caf72cd04db1c602ebd74a3db25dc6d918b9943d7/msgpack-0.6.2-cp37-cp37m-macosx_10_14_x86_64.whl (79kB)
     |████████████████████████████████| 81kB 1.1MB/s
Collecting lark-parser
  Downloading https://files.pythonhosted.org/packages/34/b8/aa7d6cf2d5efdd2fcd85cf39b33584fe12a0f7086ed451176ceb7fb510eb/lark-parser-0.7.8.tar.gz (276kB)
     |████████████████████████████████| 276kB 5.2MB/s
Collecting semver
  Downloading https://files.pythonhosted.org/packages/38/f1/0c9b852c5136364c0dd59ee1d89e0d7599bef6c9f1869e0f2e07882253ea/semver-2.9.0-py2.py3-none-any.whl
Collecting idna<2.9,>=2.5
  Downloading https://files.pythonhosted.org/packages/14/2c/cd551d81dbe15200be1cf41cd03869a46fe7226e7450af7a6545bfc474c9/idna-2.8-py2.py3-none-any.whl (58kB)
     |████████████████████████████████| 61kB 7.7MB/s
Collecting urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1
  Downloading https://files.pythonhosted.org/packages/b4/40/a9837291310ee1ccc242ceb6ebfd9eb21539649f193a7c8c86ba15b98539/urllib3-1.25.7-py2.py3-none-any.whl (125kB)
     |████████████████████████████████| 133kB 8.3MB/s
Collecting certifi>=2017.4.17
  Downloading https://files.pythonhosted.org/packages/b9/63/df50cac98ea0d5b006c55a399c3bf1db9da7b5a24de7890bc9cfd5dd9e99/certifi-2019.11.28-py2.py3-none-any.whl (156kB)
     |████████████████████████████████| 163kB 7.7MB/s
Collecting chardet<3.1.0,>=3.0.2
  Downloading https://files.pythonhosted.org/packages/bc/a9/01ffebfb562e4274b6487b4bb1ddec7ca55ec7510b22e4c51f14098443b8/chardet-3.0.4-py2.py3-none-any.whl (133kB)
     |████████████████████████████████| 143kB 7.7MB/s
Collecting six
  Downloading https://files.pythonhosted.org/packages/65/26/32b8464df2a97e6dd1b656ed26b2c194606c16fe163c695a992b36c11cdf/six-1.13.0-py2.py3-none-any.whl
Building wheels for collected packages: pyaes, lark-parser
  Building wheel for pyaes (setup.py) ... done
  Created wheel for pyaes: filename=pyaes-1.6.1-cp37-none-any.whl size=26346 sha256=8cbb34e27034534546bb94436c944ea736fcc742cda41fa9d61908223c47b399
  Stored in directory: /private/var/folders/7r/f8pztdh56yl6hd3mjzyc8cp40000gn/T/pip-ephem-wheel-cache-411o3x6b/wheels/bd/cf/7b/ced9e8f28c50ed666728e8ab178ffedeb9d06f6a10f85d6432
  Building wheel for lark-parser (setup.py) ... done
  Created wheel for lark-parser: filename=lark_parser-0.7.8-py2.py3-none-any.whl size=62514 sha256=8583dc46fe67c1f60bb69089859eafebd1a6e04bd40e14a56eb05268e12b1a9a
  Stored in directory: /private/var/folders/7r/f8pztdh56yl6hd3mjzyc8cp40000gn/T/pip-ephem-wheel-cache-411o3x6b/wheels/01/a2/30/ebae6ffa73cf3aa1c972a24d4c78388afd910f91e43bf554aa
Successfully built pyaes lark-parser
Installing collected packages: idna, urllib3, certifi, chardet, requests, six, ecdsa, base58, pyaes, msgpack, lark-parser, semver, fetchai-ledger-api
Successfully installed base58-1.0.3 certifi-2019.11.28 chardet-3.0.4 ecdsa-0.14.1 fetchai-ledger-api-0.10.2 idna-2.8 lark-parser-0.7.8 msgpack-0.6.2 pyaes-1.6.1 requests-2.22.0 semver-2.9.0 six-1.13.0 urllib3-1.25.7
Processing ./scripts/fetchai_netutils
Building wheels for collected packages: fetchai-netutils
  Building wheel for fetchai-netutils (setup.py) ... done
  Created wheel for fetchai-netutils: filename=fetchai_netutils-0.0.6a1-cp37-none-any.whl size=12854 sha256=96ecd9f36306a7c089c1cc3e3fd0c30ee7dfc1b1e0b88b27aa4fd3130ca11097
  Stored in directory: /private/var/folders/7r/f8pztdh56yl6hd3mjzyc8cp40000gn/T/pip-ephem-wheel-cache-cc6jpxhw/wheels/92/d2/31/1254dfee5f7f0d8a11db2edf6439ef6946cbfd0c0b430789e0
Successfully built fetchai-netutils
Installing collected packages: fetchai-netutils
Successfully installed fetchai-netutils-0.0.6a1

$ pip3 freeze | grep fetch
fetchai-ledger-api==0.10.2
fetchai-netutils==0.0.6a1
```

**Comment by cyenyxe** [comment]
I just looked at the Jenkins output. I think all the failed builds are due to the same error:

> Cannot uninstall requirement fetchai-ledger-api, not installed
> script returned exit code 1

Maybe it has to do with the pip version?

**Comment by n-hutton** [comment]
@cyenyxe you're right, it does fail since it is not already installed on the jenkins server (each build has a clean slate). I think that if I move the -e in the script downwards it won't cause this to fail but just issue the message

**Comment by ejfitzgerald** [comment]
@n-hutton Interesting out of interest have you tried the `-U` upgrade flag:

`pip3 install --no-cache-dir --ignore-installed -U --force-reinstall`

**Review by ejfitzgerald — APPROVED** [review]
LGTM - @wkaluza when you put the `--no-cache-dir` and `--ignore-installed` was that for a specific use case. For the cases I was using I didn't observe and difference in behaviour to simply just `--force-reinstall`

**Pull request merged**